src/main/scala/it/pagopa/interop/apigateway/api/impl/GatewayApiServiceImpl.scala

```diff
@@ -78,6 +80,27 @@ final case class GatewayApiServiceImpl(
     }
   }
 
+  override def upsertTenant(tenantSeed: TenantSeed)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerTenant: ToEntityMarshaller[Tenant]
+  ): Route = authorize {
+    logger.info(s"Upserting tenant with extenalId ${tenantSeed.externalId}")
+
+    val result: Future[Tenant] =
+      tenantProcessService.upsertTenant(m2mTenantSeedFromApi(tenantSeed)).map(m2mTenantToApi)
```

Comment:
@galales I thought that is the tenantProcess itself that performs all the checks, that's why the impl is so simple

Reply:
that's right, all business logic and authorizations are in charge of the process 👍